File: content/snippet/code-signal/arcade/intro/eruption-of-light/is-mac48-address/is-mac48-address.py
import string


def is_mac48_address(input_string):
    input_string = input_string.split('-')

    if len(input_string) != 6:
        return False

    for i in input_string:
        if len(i) != 2 or (not all(c in string.hexdigits for c in i)):
            return False

    return True


print(is_mac48_address('00-1B-63-84-45-E6'))
print(is_mac48_address('ZZ-1B-63-84-45-Z6'))
print(is_mac48_address('not a mac-address'))

File: content/snippet/code-signal/arcade/intro/eruption-of-light/is-mac48-address/is-mac48-address.js
function isMAC48Address(inputString) {
  const regex = /^[0-9A-F]{2}(-[0-9A-F]{2}){5}$/;
  return regex.test(inputString);
}

console.log(isMAC48Address('00-1B-63-84-45-E6'));
